src/versionbits.h

```diff
@@ -17,24 +17,36 @@ static const int32_t VERSIONBITS_TOP_MASK = 0xE0000000UL;
 /** Total bits available for versionbits */
 static const int32_t VERSIONBITS_NUM_BITS = 29;
 
+/** BIP 9 defines a finite-state-machine to deploy a softfork in multiple stages.
+ *  It relies on the chain being a monotonic clock to start or timeout the
+ *  deployment. State transitions happen during retarget period if conditions
+ *  are met. In case of reorg, transitions can go backward. Without transition,
+ *  state is inherited between periods. All blocks of a period share the same state.
+ */
 enum class ThresholdState {
-    DEFINED,
-    STARTED,
-    LOCKED_IN,
-    ACTIVE,
-    FAILED,
+    DEFINED,   // First state that each softfork starts out as. The genesis block is by definition in this state for each deployment.
```

Comment:
These match the `States` definitions in [BIP9](https://github.com/bitcoin/bips/blob/master/bip-0009.mediawiki#states) 👍